src/internal/connector/onedrive/restore.go

```diff
@@ -390,6 +381,155 @@ func RestoreCollection(
 	}
 }
 
+type fileFetcher interface {
+	Fetch(ctx context.Context, name string) (data.Stream, error)
+}
+
+func restoreV1File(
+	ctx context.Context,
+	source driveSource,
+	service graph.Servicer,
+	drivePath *path.DrivePath,
+	fetcher fileFetcher,
+	restoreFolderID string,
+	copyBuffer []byte,
+	parentPerms []UserPermission,
+	permissionIDMappings map[string]string,
+	restorePerms bool,
+	itemData data.Stream,
+) (details.ItemInfo, error) {
+	trimmedName := strings.TrimSuffix(itemData.UUID(), DataFileSuffix)
+
+	itemID, itemInfo, err := restoreData(
+		ctx,
+		service,
+		trimmedName,
+		itemData,
+		drivePath.DriveID,
+		restoreFolderID,
+		copyBuffer,
+		source)
+	if err != nil {
+		return details.ItemInfo{}, err
+	}
+
+	// Mark it as success without processing .meta
+	// file if we are not restoring permissions
+	if !restorePerms {
+		return itemInfo, nil
+	}
+
+	// Fetch item permissions from the collection and restore them.
+	metaName := trimmedName + MetaFileSuffix
+
+	permsFile, err := fetcher.Fetch(ctx, metaName)
+	if err != nil {
+		err = clues.Wrap(err, "getting item metadata")
```

Comment:
I know this is code that I had written, but now that I look at it, having the name available in the error might end up being useful.